storage/storage-samples/storage-owner-sample/src/main/java/org/fido/iot/storage/OwnerDbManager.java

```diff
@@ -280,6 +298,169 @@ public void assignSviToDevice(DataSource ds, UUID guid, String sviString) {
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+
+    return isWgetContentVerificationEnabled;
+  }
+
+  /**
+   * Calculated sha384 hash of file content to be transferred using wget:url. Adds the hash entry to
+   * OWNER_SERVICEINFO table Includes appropriate mapping in the sviString for corresponding hash
+   * file.
+   *
+   * @param ds datasource
+   * @param sviString svi mapping string to be updated
+   * @return updated sviString with activate module entries
+   */
+  public String insertWgetContentHash(DataSource ds, String sviString) {
+    StringBuilder sb = new StringBuilder();
+    List<String> sviEntries = Arrays.asList(sviString.split(","));
+    List<String> newSviEntries = new ArrayList<>();
+
+    for (String sviEntry : sviEntries) {
+      String[] sviEntryArray = sviEntry.split(SVI_ENTRY_DELIMETER);
+
+      // Get the filename whose hash needs to be calculated.
+      if (sviEntryArray[0].trim().equals("sdo_wget:filename")) {
+        String filename = null;
+        String sviId = null;
+        byte[] sviContent = null;
+
+        String sql = "SELECT CONTENT FROM OWNER_SERVICEINFO WHERE SVI_ID = ?";
+        try (Connection conn = ds.getConnection();
+            PreparedStatement pstmt = conn.prepareStatement(sql)) {
+          pstmt.setString(1, sviEntryArray[1]);
+          try (ResultSet rs = pstmt.executeQuery()) {
+            if (rs.next()) {
+              filename = new String(rs.getBytes(1), StandardCharsets.UTF_8);
+              // concatenate the filename and the hash to generate a unique SVI_ID.
+              sviId = new StringBuilder(filename + "_hash").toString();
+            }
+          }
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+
+        // Read the file content whose hash needs to be calculated.
+        sql = "SELECT CONTENT FROM OWNER_SERVICEINFO WHERE SVI_ID = ?";
+        try (Connection conn = ds.getConnection();
+            PreparedStatement pstmt = conn.prepareStatement(sql)) {
+          pstmt.setString(1, filename);
+          try (ResultSet rs = pstmt.executeQuery()) {
+            if (rs.next()) {
+              sviContent = rs.getBytes(1);
+            }
+          }
+        } catch (SQLException e) {
+          throw new RuntimeException(e);
+        }
+
+        // calculate SHA384 hash of the file content.
+        byte[] wgetFileContentHash =
+            (new CryptoService().hash(Const.SHA_384, sviContent)).getAsBytes(Const.HASH);
+
+        // Adds the hash entry to OWNER_SERVICEINFO table.
+        addServiceInfo(ds, sviId, wgetFileContentHash, false);
+
+        // add the mapping to GUID_OWNERSVI
+        newSviEntries.add(("sdo_wget:sha-384=" + sviId));
+      }
+    }
+
+    int i = 0;
+    for (String sviEntry : sviEntries) {
+      sb.append(sviEntry + ",");
+      String[] sviEntryArray = sviEntry.split(SVI_ENTRY_DELIMETER);
+      if (sviEntryArray[0].trim().equals("sdo_wget:url")) {
```

Comment:
consider using a constant for this literal

Reply:
Done